src/components/hooks/use-livechat.js

```diff
@@ -126,7 +126,7 @@ export const useLivechat = () => {
                         }
                     } else {
                         if (window.LiveChatWidget.get('chat_data')) {
-                            const chatID = window.LiveChatWidget.get('chat_data').chatId
+                            const chatID = window.LiveChatWidget.get('chat_data').chatId ?? ''
```

Comment:
```suggestion
                            const chatID = window.LiveChatWidget.get('chat_data')?.chatId
```

Reply:
Could we use optional chaining here instead?

Reply:
I doubt we can actually...cause optional chaining would mean that there is a possibility that nothing is passed. From what i understand there was an error on console because nothing was passed for chatID. Maybe @kevinw-binary can confirm on this?

Reply:
There's a chance that `window.LiveChatWidget.get('chat_data')` isn't  defined yet which will cause the error

or is it better to implement `try and catch` since this code is using API to get the data ?

Reply:
I guess the current logic is correct with a fallback, since we have to assign a value to  chat_id, so `chat_id: undefined` is not an option.

But we need to make sure that `chat_id: ''` is acceptable @kevinw-binary please confirm 🙏🏻

Reply:
I think it will catch the error on the line below here  

`customerSDK?.deactivateChat({ chatId: chatID }).catch((error) => { `

and the fallback will supposedly fix the error from trackJS ?

Reply:
I believe that the issue appears because we have [`console.error`](https://github.com/binary-com/deriv-com/pull/1391/files#diff-e5e90c60d24e7dae6576024383c266127f2d7aef9cb1717329e7e557cc98fe9fR132) 

A fallback should prevent going to `catch` block, but I am not very sure if an empty string would do it.

Reply:
As per documentation [`chatId`](https://developers.livechat.com/docs/extending-chat-widget/customer-sdk/#deactivatechat) needs to be of type `string` therefore it's better to have `''` as a fallback to avoid `chatId: undefind`

Reply:
either that, or call deactivateChat if only there is valid chatId